### PR TITLE
Add internal cs name to vm during the ingest

### PR DESCRIPTION
**Pull request by DK101010**
### Description
Each VM deployed in CS get an alphanumeric value as  internal CS name (cloud.vm.internal.name) . Each VM that will be ingested get no internal CS name and the internal name = VM name. 

CS use for a couple of actions the internal CS name if exist and therefore CS can no longer update this properties. For an example power state.

This PR add the internalCSName to VirtualMaschineMO and UnmanagedInstanceTO  inside the loadCache method to use this value  as instance name during the ingest. 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?
Tested in own VMWare env.

### CODE REVIEW

**Review by DaanHoogland — APPROVED** [review]
makes sense, clgtm

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@DK101010 can you please merge the latest `main` and fix the build issue
```
[ERROR] /jenkins/workspace/acs-centos7-pkg-builder/dist/rpmbuild/BUILD/cloudstack-4.16.0.0-SNAPSHOT/server/src/main/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImpl.java:[961,11] error: reference to StringUtils is ambiguous
[ERROR]   both class org.apache.commons.lang3.StringUtils in org.apache.commons.lang3 and class org.apache.commons.lang.StringUtils in org.apache.commons.lang match
```

**Comment by shwstppr** [comment]
@blueorangutan package

**Review by shwstppr — APPROVED** [review]
Works when VM has internal name set and it is of format `i-[ACCOUNT_ID]-[VM_ID]-`

**Comment by DaanHoogland** [comment]
@shwstppr did you try other formats?

**Comment by DaanHoogland** [comment]
And, @shwstppr , what do you mean "when VM has internal name"? isn't it set during ingestion?

**Comment by shwstppr** [comment]
@DaanHoogland 

- I've not debugged the reason but not all VMs have an internal name in vCenter in my env
- I cloned a VM in vCenter. Cloned VM didn't have an internal name assigned though it had the property. 
Screenshot from 2021-08-02 14-47-17

- I manually set the value there to `ak-cloned`
Screenshot from 2021-08-02 14-48-45

- Code didn't pick this value during import as the method `isUserVMInternalCSName` would return false. So after import `instance_name` in the vm_instance` table for the VM remained `cloned`
Screenshot from 2021-08-02 13-51-36
Screenshot from 2021-08-02 14-01-04

**Comment by shwstppr** [comment]
@DaanHoogland I think @DK101010 wants to fix the case where a CloudStack VM is unmanaged and then imported again it gets the correct instance name. In that case unmanaged VM will already have an internal name set. During import no change is made to `cloud.vm.internal.name` property

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-65u2

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-65u2

**Review by nvazquez — APPROVED** [review]
Code LGTM

File: vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/HostMO.java [hunk 1]
```diff
@@ -579,8 +579,9 @@ private void loadVmCache() throws Exception {
 
                     if (s_logger.isTraceEnabled())
                         s_logger.trace("put " + vmName + " into host cache");
-
-                    _vmCache.put(vmName, new VirtualMachineMO(_context, oc.getObj()));
+                    VirtualMachineMO virtualMaschine = new VirtualMachineMO(_context, oc.getObj());
```
Comment (sureshanaparti) [inline vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/HostMO.java:582]: typo

```suggestion
                    VirtualMachineMO virtualMachine = new VirtualMachineMO(_context, oc.getObj());
```

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@blueorangutan test keepEnv

**Pull request merged**